polyfill/lib/zoneddatetime.mjs

```diff
@@ -298,7 +298,15 @@ export class ZonedDateTime {
   }
   startOfDay() {
     if (!ES.IsTemporalZonedDateTime(this)) throw new TypeError('invalid receiver');
-    throw new Error('startOfDay() not implemented yet');
+    const dt = dateTime(this);
+    const DateTime = GetIntrinsic('%Temporal.DateTime%');
+    const dtStart = new DateTime(GetSlot(dt, ISO_YEAR), GetSlot(dt, ISO_MONTH), GetSlot(dt, ISO_DAY), 0, 0, 0, 0, 0, 0);
```

Comment:
For non-ISO time calendars, is all-zero time ISO time units guaranteed to be the start of the day?  If so then this code is good. If not, then do you need to ask the calendar when the start of the day is?

Reply:
I believe that this is OK because we stipulated that non-ISO calendars must still define days to be the same 24-hour periods as the ISO calendar.